src/visualization/components/Player.tsx

```diff
@@ -87,45 +121,252 @@ class Player extends React.Component<any, any> {
     }
 
     playImpression(index) {
-        this.props.selectImpression(this.props.playlist[index]);
+        this.props.selectImpression(this.props.session[index]);
+    }
+
+    uploadAndShare() {
```

Comment:
>uploadAndShare [](start = 4, length = 14)

I feel like the uploading and sharing functionality should move into a separate component